include/range/v3/view/reverse.hpp

```diff
@@ -107,22 +107,21 @@ namespace ranges
             {
                 return ranges::size(rng_);
             }
-
-            using unreverse_me = reverse_view;
         };
 
-        template<typename Rng_>
-        struct reverse_view<Rng_, true>
-          : identity_adaptor<decltype(std::declval<Rng_ &>().base())>
+#if !defined(__GNUC__) || defined(__clang__) || __GNUC__ >= 5
+        template<typename Rng>
+        struct reverse_view<reverse_view<Rng>>
+          : identity_adaptor<Rng>
```

Comment:
This partial specialization is buggy. For every `reverse_view<T>`, `.base()` should return a `T`. So `reverse_view<reverse_view<Rng>>::base()` should return a `reverse_view<Rng>`. But this specialization, which gets its `base()` from `identity_adaptor<Rng>` _doesn't_ return a `reverse_view<Rng>`; it returns a `Rng`. This specialization of `reverse_view` needs to define its own `base()` that reconstitutes a `reverse_view<Rng>` from the `Rng` object -- from which it should inherit directly.

So two things:
1. Implementing `reverse_view<reverse_view<Rng>>` in terms of `identity_adaptor<Rng>` is wrong.
2. `identity_adaptor<Rng>::base()` is highly suspect and should be removed.

Reply:
Yes, I've discovered how fragile this is; I'm planning to revert to the "weird customizations happen in `view::foo` design."